spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java

```diff
@@ -87,6 +92,11 @@ public ClientHttpResponse intercept(final HttpRequest request, final byte[] body
 				!lbProperties.isEnabled() || retryPolicy == null ? new NeverRetryPolicy()
 						: new InterceptorRetryPolicy(request, retryPolicy, loadBalancer,
 						serviceName));
+		final RetryListener[] retryListeners = this.retryListenerFactory.createRetryListeners(serviceName);
```

Comment:
I dont think it is necessary to create the listeners on every intercept call, it is probably enough to create them once.

Reply:
@ryanjbaxter Thanks for helping me review the code.
Read `RetryLoadBalancerInterceptor#intercept()` source code:
`RetryTemplate template = this.retryTemplate == null ? new RetryTemplate() : this.retryTemplate;`
I found that each interception call will be re-creates a `RetryTemplate` (since `this.retryTemplate` is always null for every request coming in) so I will create listeners every time.
I understand so right? Hope to receive your reply. Thanks again.

Reply:
Yeah the service name can change, this is fine then.